sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientLiveTests.cs

```diff
@@ -895,16 +895,16 @@ await using (var consumer = new EventHubConsumerClient(EventHubConsumerClient.De
                     var partition = (await consumer.GetPartitionIdsAsync(cancellationSource.Token)).First();
                     await SendEventsAsync(connectionString, sourceEvents, new CreateBatchOptions { PartitionId = partition }, cancellationSource.Token);
 
-                    // Read the events and validate the resulting state.
-
                     // Create a local function that will close the consumer after five events have
-                    // been read.
+                    // been read.  Because the close happens in the middle of iteration, allow for a short
+                    // delay to ensure that the state transition has been fully captured.
 
                     async Task<bool> closeAfterRead(ReadState state)
                     {
                         if (state.Events.Count >= 2)
                         {
-                            await consumer.CloseAsync(cancellationSource.Token).ConfigureAwait(false);
+                            await consumer.CloseAsync(cancellationSource.Token);
+                            await Task.Delay(250);
```

Comment:
Just curious - Is an actual delay necessary here, or is a `Task.Yeild()` enough?

Reply:
I would imagine that yielding would be enough, since we just want to ensure that the state updates, but I always seem to forget that exists and `Delay(0)` just reads weird to me.    I'll take a note to revisit, but will probably avoid pushing up changes if the live suite passes this way so that I can unblock tomorrow's run.